autogpt_platform/backend/backend/blocks/ayrshare/post.py

```diff
@@ -0,0 +1,531 @@
+        )
+        response = client.create_post(
+            post=input_data.post,
+            platforms=platforms,
+            media_urls=input_data.media_urls,
+            is_video=input_data.is_video,
+            schedule_date=iso_date,
+            disable_comments=input_data.disable_comments,
+            shorten_links=input_data.shorten_links,
+            unsplash=input_data.unsplash,
+            requires_approval=input_data.requires_approval,
+            random_post=input_data.random_post,
+            random_media_url=input_data.random_media_url,
+            notes=input_data.notes,
+            profile_key=profile_key,
+        )
+        return response
+
+    def run(
+        self,
+        input_data: "AyrsharePostBlockBase.Input",
+        *,
+        profile_key: SecretStr,
+        **kwargs,
+    ) -> BlockOutput:
+        """Run the block."""
+        platforms = [SocialPlatform.FACEBOOK]
+
+        if not profile_key:
+            yield "error", "Please Link a social account via Ayrshare"
+            return
+
+        post_result = self._create_post(
+            input_data, platforms=platforms, profile_key=profile_key.get_secret_value()
+        )
+        if isinstance(post_result, PostError):
+            yield "error", post_result.message
+            return
+        yield "post_result", post_result
+
+
+class PostToFacebookBlock(AyrsharePostBlockBase):
+    """Block for posting to Facebook."""
+
+    def __init__(self):
+        super().__init__(
+            id="3352f512-3524-49ed-a08f-003042da2fc1",
+            description="Post to Facebook using Ayrshare",
+        )
+
+    def run(
+        self,
+        input_data: AyrsharePostBlockBase.Input,
+        *,
+        profile_key: SecretStr,
+        **kwargs,
+    ) -> BlockOutput:
+        """Post to Facebook."""
+        if not profile_key:
+            yield "error", "Please Link a social account via Ayrshare"
+            return
+
+        post_result = self._create_post(
+            input_data,
+            [SocialPlatform.FACEBOOK],
+            profile_key=profile_key.get_secret_value(),
+        )
+        if isinstance(post_result, PostError):
+            yield "error", post_result.message
+            return
+        yield "post_result", post_result
+
+
+class PostToXBlock(AyrsharePostBlockBase):
+    """Block for posting to X / Twitter."""
+
+    def __init__(self):
+        super().__init__(
+            id="9e8f844e-b4a5-4b25-80f2-9e1dd7d67625",
+            description="Post to X / Twitter using Ayrshare",
+        )
+
+    def run(
+        self,
+        input_data: AyrsharePostBlockBase.Input,
+        *,
+        profile_key: SecretStr,
+        **kwargs,
+    ) -> BlockOutput:
+        """Post to Twitter."""
+        if not profile_key:
+            yield "error", "Please Link a social account via Ayrshare"
+            return
+
+        post_result = self._create_post(
+            input_data,
+            [SocialPlatform.TWITTER],
```

Comment:
if this is the only difference between all these blocks, this can be a class property and then `run` can be implemented on the base class instead of on all these subclasses